docs/pages/api.rst

```diff
@@ -48,14 +48,38 @@ Signature: ``Collection::fromIterable(string $filepath): Collection;``
 
     Collection::fromFile('http://loripsum.net/api');
 
+fromGenerator
+~~~~~~~~~~~~~
+
+Create a collection from a `Generator`_.
+
+.. warning:: The difference between this constructor and ``fromIterable`` is that
+    the generator is decorated with a caching Iterator. ``Generators`` are not
+    **rewindable** by design and using ``fromIterable`` automatically adds the
+    caching layer for you.
+
+.. tip:: You can reproduce the same behavior by using ``fromIterable`` directly
+    followed by the ``cache`` operation.
+
+Signature: ``Collection::fromGenerator(Generator $generator): Collection;``
+
+.. code-block:: php
+
+    $generator = (static fn () => yield from range(1, 10))();
+    $generator->next();
+    $generator->next();
+
+    $collection = Collection::fromGenerator($generator);
```

Comment:
this example doesn't tell me much :(. Can we show how this collection will iterate? For example by applying `all` or using a `foreach` loop?

Reply:
Work complete! https://www.youtube.com/watch?v=PoMhMZZjbGg